### PR TITLE
[MAD][DATA1018][data-analysis-with-pandas] Ivan Cernicharo

### PR DESCRIPTION
**Pull request by ijcernicharo**
Good morning Belen,
Hope you are great and eager to make many corrections. I did not have many problems as far as cleaning the data and normalizing it. I look forward to hearing from your feedback.
The last commit has been data-analysis-with-pandas MOD #3 FINAL - FINISHED.
Best regards,
Ivan

### CODE REVIEW

**Comment by ijcernicharo** [comment]
Good Morning Belen,
I have done some modifications in my notebook. Now the last update is MOD #4 This is the FINAL on - No kidding. I look forward to hearing from your feedback.
Best regards,
Ivan

File: module-2/data-analysis-with-pandas/your-code/data analysis with pandas.ipynb [hunk 1]
```diff
@@ -97,9 +97,9 @@
     "crashes_raw=crashes_raw.drop('Summary',axis=1)\n",
```
Comment (Gobinde43) [inline module-2/data-analysis-with-pandas/your-code/data analysis with pandas.ipynb]: - There are 2 Python modules that let you converting number to name and viceversa in case of calendar dates. One is calendar module (just typing import calendar) and the other strptime (I think in time within strptime). Take a look on them. Using this useful modules save you time creating a dictionary. 
```Python
# I guess is something like the following 
calendar.month_abbr[month_number]
```
- Before using describe method, it's a good practice to convert all data types to the correct ones (info() method always after importing!!). This way describe only get statistics summary for "real" numeric variables. 
- You can also import pandas_profiling and do something like the following: 
```Python
pandas_profiling.ProfileReport(your_dataset_name)
```
Let's try on Monday!!
- It's always a good practice to print the head of your dataframe as well as to include a brief description of the meaning of each column before you run your analysis process. 
- Let's talk on Monday about the correlation analysis you have done. It's a good practice to get a complete correlation matrix for all your numeric variable in your dataset, don't chunk the analysis. If possible try  build a correlation matrix like a heatmap. 
I have this chunk of code that you can check it works for you. 
```Python
import seaborn as sns
cmap = cmap=sns.diverging_palette(5, 250, as_cmap=True)

def magnify():
    return [dict(selector="th",
                 props=[("font-size", "7pt")]),
            dict(selector="td",
                 props=[('padding', "0em 0em")]),
            dict(selector="th:hover",
                 props=[("font-size", "12pt")]),
            dict(selector="tr:hover td:hover",
                 props=[('max-width', '200px'),
                        ('font-size', '12pt')])
]

wine_corr.style.background_gradient(cmap, axis=1)\
    .set_properties(**{'max-width': '80px', 'font-size': '10pt'})\
    .set_caption('Wine feature Correlation')\
    .set_precision(2)\
    .set_table_styles(magnify())
```
- Variables standarization is a very used method when we have a clear idea about which our target variable will be in a further modelling process. I think the better way of doing this using scikit-learn. 
```Python
from sklearn.preprocessing import StandardScaler
```
